### DISCUSSION TITLE
Get all the anchors defined in a glyph

### DISCUSSION THREAD

**Discussion by murasu** [harfbuzz/harfbuzz#2864] [Q&A] [answered]
Is there an API or possible steps I can follow to get all the anchor points defined in a glyph?

**Comment by behdad** [comment, marked as answer]
Currently that's not part of our API and not planned to be. Simply because it's a lot of API to expose such features and only useful to font development / QA workflows.

Normally we send people to fonttools for such requests.

That said, I do have a longterm plan to expose much of the internals as C++ API, such that other apps can use our font loading infrastructure and table implementation. But that's way into the future, since we would need to separate the internal C++ headers into publicly-exposed (with some API compatibility guarantee) versus plain internal implementation details. Currently we don't have that distinction.

If you cannot use fonttools and you are building HarfBuzz yourself, I suggest you include and access internal structures. That's not something I've ever encouraged anyone to do though. :)
